### ISSUE TITLE
IntellijCoder plugin imports src without test cases [throws IndexNotReadyException]

### ISSUE THREAD

**Issue by sameer2086** [fadeyev/intellijcoder#11]
I am using IntelliJ CE 14.1.2, jdk 8 and latest version of IntellijCoder plugin. I am running the IDE on Mac OS X 10.10.2. The issue manifests frequently when I try to open a problem from IDE launched Topcoder applet. IntelliJ logs following event and only imports src files without test cases. 

```java
Please change caller according to com.intellij.openapi.project.IndexNotReadyException documentation
com.intellij.openapi.project.IndexNotReadyException: Please change caller according to com.intellij.openapi.project.IndexNotReadyException documentation
	at com.intellij.util.indexing.FileBasedIndexImpl.handleDumbMode(FileBasedIndexImpl.java:854)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:803)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:787)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:250)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:238)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:76)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:95)
	at com.intellij.psi.stubs.StubIndexImpl.get(StubIndexImpl.java:227)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:144)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:134)
	at com.intellij.psi.impl.java.stubs.index.JavaFullClassNameIndex.get(JavaFullClassNameIndex.java:48)
	at com.intellij.psi.impl.file.impl.JavaFileManagerImpl.findClassInIndex(JavaFileManagerImpl.java:145)
	at com.intellij.psi.impl.file.impl.JavaFileManagerImpl.findClass(JavaFileManagerImpl.java:129)
	at com.intellij.psi.impl.PsiElementFinderImpl.findClass(PsiElementFinderImpl.java:54)
	at com.intellij.psi.impl.JavaPsiFacadeImpl.findClass(JavaPsiFacadeImpl.java:99)
	at com.intellij.psi.impl.PsiElementFactoryImpl.createTypeByFQClassName(PsiElementFactoryImpl.java:411)
	at com.intellij.psi.PsiType.getTypeByName(PsiType.java:147)
	at com.intellij.psi.PsiType.getJavaLangObject(PsiType.java:159)
	at com.intellij.psi.impl.PsiClassImplUtil.getSuperTypes(PsiClassImplUtil.java:902)
	at com.intellij.psi.impl.source.PsiClassImpl.getSuperTypes(PsiClassImpl.java:249)
	at com.intellij.psi.impl.PsiClassImplUtil.processSuperTypes(PsiClassImplUtil.java:774)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClassNotCached(PsiClassImplUtil.java:709)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClass(PsiClassImplUtil.java:478)
	at com.intellij.psi.impl.PsiClassImplUtil.processDeclarationsInClass(PsiClassImplUtil.java:448)
	at com.intellij.psi.impl.source.PsiClassImpl.processDeclarations(PsiClassImpl.java:488)
	at com.intellij.psi.scope.util.PsiScopesUtil.treeWalkUp(PsiScopesUtil.java:72)
	at com.intellij.psi.scope.util.PsiScopesUtil.treeWalkUp(PsiScopesUtil.java:54)
	at com.intellij.psi.scope.util.PsiScopesUtil.resolveAndWalk(PsiScopesUtil.java:215)
	at com.intellij.psi.scope.util.PsiScopesUtil.resolveAndWalk(PsiScopesUtil.java:146)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl.resolve(PsiJavaCodeReferenceElementImpl.java:438)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl.access$100(PsiJavaCodeReferenceElementImpl.java:55)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl$OurGenericsResolver.resolve(PsiJavaCodeReferenceElementImpl.java:339)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl$OurGenericsResolver.resolve(PsiJavaCodeReferenceElementImpl.java:331)
	at com.intellij.psi.impl.source.resolve.ResolveCache$4.compute(ResolveCache.java:191)
	at com.intellij.psi.impl.source.resolve.ResolveCache$4.compute(ResolveCache.java:188)
	at com.intellij.openapi.util.RecursionManager$2.doPreventingRecursion(RecursionManager.java:112)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:188)
	at com.intellij.psi.impl.PsiImplUtil.multiResolveImpl(PsiImplUtil.java:767)
	at com.intellij.psi.impl.PsiImplUtil.multiResolveImpl(PsiImplUtil.java:757)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl.multiResolve(PsiJavaCodeReferenceElementImpl.java:366)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl.advancedResolve(PsiJavaCodeReferenceElementImpl.java:359)
	at com.intellij.psi.impl.source.PsiJavaCodeReferenceElementImpl.resolve(PsiJavaCodeReferenceElementImpl.java:328)
	at com.intellij.psi.impl.source.tree.JavaTreeCopyHandler.encodeInformationInRef(JavaTreeCopyHandler.java:207)
	at com.intellij.psi.impl.source.tree.JavaTreeCopyHandler.encodeInformation(JavaTreeCopyHandler.java:141)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:62)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:69)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:53)
	at com.intellij.psi.impl.source.tree.ChangeUtil.encodeInformation(ChangeUtil.java:47)
	at com.intellij.psi.impl.file.PsiDirectoryImpl.updateAddedFile(PsiDirectoryImpl.java:392)
	at com.intellij.psi.impl.file.PsiDirectoryImpl.add(PsiDirectoryImpl.java:456)
	at com.intellij.psi.impl.file.PsiJavaDirectoryImpl.add(PsiJavaDirectoryImpl.java:64)
	at intellijcoder.idea.IntelliJIDEA$ModuleCreator.createJavaFile(IntelliJIDEA.java:359)
	at intellijcoder.idea.IntelliJIDEA$ModuleCreator.create(IntelliJIDEA.java:334)
	at intellijcoder.idea.IntelliJIDEA.createModule(IntelliJIDEA.java:155)
	at intellijcoder.idea.IntelliJIDEA.access$100(IntelliJIDEA.java:45)
	at intellijcoder.idea.IntelliJIDEA$1$1.run(IntelliJIDEA.java:60)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:937)
	at intellijcoder.idea.IntelliJIDEA$1.run(IntelliJIDEA.java:57)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:332)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:209)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:715)
	at java.awt.EventQueue.access$400(EventQueue.java:82)
	at java.awt.EventQueue$2.run(EventQueue.java:676)
	at java.awt.EventQueue$2.run(EventQueue.java:674)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:86)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:685)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:734)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:569)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
```

**Comment by pfilippi24** [comment]
I get the same issue with Intellij Ultimate 14.1.2 and jdk 8 running on Mint 17.

I already identified the error and will submit a pull request for this, if I can fix it.

**Comment by sameer2086** [comment]
Thanks for your help!

**Comment by Vennik** [comment]
Same issue on Intellij Ultimate 14.1.3 and jdk 8 running on Windows 8.1.

**Comment by silkyprogrammer** [comment]
Is this problem fixed? I am getting same error not sure about how to solve it, any help appreciated.

**Comment by fadeyev** [comment]
I'm going to merge fix for the issue shortly

**Comment by silkyprogrammer** [comment]
Thank you @fadeyev - I noticed that, when I upgraded to IntelliJ CE 14.1.4 (IC-141.1532.4) yesterday this problem disappeared.

**Comment by muztaba** [comment]
I am using IntelliJ CE 14.1.4 (IC-141.1532.4) also. But the problem still there.

**Comment by vkuri** [comment]
Using IntelliJ CE 14.1.4. Facing the exact same issue. The tests are not getting imported automatically.

**Comment by fadeyev** [comment]
The fix developed by @mmicheo has been released in 2.1 version and will be available as soon as JetBrains checks it (1-2 days)

**Comment by sameer2086** [comment]
Thanks for fixing the issue. Appreciate it! 
I updated the plugin to the latest version and didn't encounter the issue yet.
